Allow disabling remote method after initialize explorer

supasate opened the pull request:
This PR is for strongloop/loopback#686 to make the explorer registering a remoteMethodDisabled event that is fired from PR strongloop/loopback#2266. The explorer will update a swagger object when getting this event to hide the disabled remote method.

slnode commented:
Can one of the admins verify this patch? To accept patch and trigger a build add comment ".*ok\W+to\W+test.*"

superkhau commented on test/explorer.test.js:
```diff
@@ -283,6 +283,37 @@ describe('explorer', function() {
+            var paths = Object.keys(res.body.paths);
+            expect(paths).to.not.contain('/products/findOne');
+          });
+        done();
```
move this done after the above expect.

```
expect(paths)...
done();
```

superkhau commented:
@supasate One thing to fix, otherwise LGTM.

superkhau commented on index.js:
```diff
@@ -104,6 +104,12 @@
 
+  // listening to remoteMethodDisabled event for updating the swaggerObject
+  // when a remote method is disabled and remote it from the Swagger UI.
+    loopbackApplication.on('remoteMethodDisabled', function() {
```
Indent comment to line up with `loopbackApplication...`

supasate replied:
@superkhau  It seems like this whole file uses this style of indentation. I'm not sure it's the convention or not, so, I just followed. Should I change the indentation of the whole file?

superkhau replied:
Hm...its probably the linter that messed up. ;( Could you add a separate commit for the indentation fixes? Much appreciated. ;)

supasate commented:
@superkhau I've fixed them and squashed the commits.

Anyway, currently, I just use the loopback dependency directly from strongloop/loopback#2266. What version of loopback in package.json should be if it's not tagged as a new version yet?

bajtos commented:
The patch contains quite few unrelated indentation changes. I'd like to wait with landing it until #158 is landed first.

bajtos commented:
Hey @supasate, sorry for the trouble, could you please rebase your patch on top of the current master? `git rebase master && git push -f`.

superkhau commented:
>The patch contains quite few unrelated indentation changes. I'd like to wait with landing it until #158 is landed first.

I ask him to do it in a separate commit (in one of the comments) so I can review it easier.

@supasate Please resolve the conflicts now that #158 has been merged and we'll get it merged in.

supasate commented:
@bajtos @superkhau Should be alright now.

superkhau commented:
LGTM. Thanks for the contribution! ;)
